### PR TITLE
Change "Quickstart" to "Demo" in navigation

### LINKED ISSUE #1605: armadaproject.io: remove references to the term "quickstart", replace with "Demo" -> link to AWS demo site

### PR DESCRIPTION
**Pull request by suprjinx**
Fixes #1605

### CODE REVIEW

**Comment by jayofdoom** [comment]
I tested this, it looked like it worked good.

File: demo.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Armada Demo
+
+This guide will show you how to take a quick test drive of an Armada
+instance already deployed to AWS EKS.
+
+## EKS
+
+The EKS demo of Armada can be found at these URLs:
+- [API Server](https://armada.demo.armadaproject.io)
```
Comment (kannon92) [inline demo.md:9]: We probably don't need this.  That link doesn't really provide anything useful so people will click and tell us the link is bad.  But not sure what the point of hyperlinking a api server without swagger docs?

Comment (suprjinx) [reply demo.md:9]: yes good point -- I'll change it in the other PR -- this file gets clobbered from master on every build

Comment (suprjinx) [reply demo.md]: well changed here, too, but master will always clobber

**Review by kannon92 — APPROVED** [review]
This looks good to me but I don’t understand what we do with the markdown

**Pull request merged**